eng/common/templates/jobs/build-images.yml

```
@@ -50,14 +40,29 @@ jobs:
   - template: ${{ format('../steps/init-docker-{0}.yml', parameters.dockerClientOS) }}
     parameters:
       setupRemoteDockerServer: ${{ parameters.useRemoteDockerServer }}
+  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+    - script: >
+        echo "##vso[task.setvariable variable=imageBuilderBuildArgs]
+        --image-info-output-path "$(imageBuilderArtifactsPath)/$(legName)-image-info.json"
+        --registry-override $(acr.server)
+        --repo-prefix $(stagingRepoPrefix)
+        --push
+        --username $(acr.userName)
+        --password $(BotAccount-dotnet-docker-acr-bot-password)
+        $(imageBuilder.queueArgs)"
+      displayName: Define imageBuilderBuildArgs Variable
+  - ${{ if eq(variables['System.TeamProject'], 'public') }}:
```

This 'if' could simply be removed and the common `$(imageBuilder.queueArgs)` could be added to the very end of the `Build Images` step.  The benefit is it gets all of the common args declared in one spot.

---

The downside to that is that imageBuilderBuildArgs would be an undeclared variable in the public build.  That produces an ignored error in Windows I think (or maybe it's Linux).  I want to avoid that because I don't like seeing such errors in the build output.

---

Ok that makes sense.